test.py

```diff
@@ -139,5 +139,21 @@ def test_size_2(self):
       metric.update_state(y_true, y_pred)
       self.assertAlmostEqual(1-3/5, metric.result().numpy())
 
+class DataTestCase(unittest.TestCase):
+    def setUp(self):
+        FLAGS(['test'])
+
+    def test_grid(self):
+      n = FLAGS.board_size
+      g = FLAGS.grid_size
+      X = np.random.rand(1, 1, n, n, 1)
+      y = np.zeros_like(X)
+      for i in range(n):
+        for j in range(n):
+          a = i // g
+          b = j // g
+          y[0, 0, i, j, 0] = np.mean(X[0, 0, a * g:(a + 1) * g, b * g:(b + 1) * g, 0])
+      self.assertIsNone(np.testing.assert_array_almost_equal(y, num_black_cells_in_grid(X)))
```

Comment:
I don't think the assertIsNone is necessary, cause the assert_array_almost_equal will raise an error if they don't match.